crates/kitsune_p2p/types/src/metrics.rs

```diff
@@ -393,6 +395,44 @@ impl MetricRecordKind {
     }
 }
 
+type WriteLenCb = Box<dyn Fn(&'static str, usize) + 'static + Send + Sync>;
+
+/// Metrics callback manager to be injected into the endpoint
+pub struct Tx2ApiMetrics {
+    write_len: Option<WriteLenCb>,
+}
+
+impl Default for Tx2ApiMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Tx2ApiMetrics {
```

Comment:
What are these still needed for?

Reply:
Kitsune internally uses these tx2 metrics structs. I'd like to refactor that, but it doesn't belong in this PR and isn't really a priority.